kyuubi-server/src/main/scala/org/apache/kyuubi/engine/flink/FlinkProcessBuilder.scala

```diff
@@ -213,6 +254,32 @@ class FlinkProcessBuilder(
     }
   }
 
+  @volatile private var tokenTempDir: java.nio.file.Path = _
+  private def generateTokenFile(): Option[(String, String)] = {
+    // We disabled `hadoopfs` token service, which may cause yarn client to miss hdfs token.
+    // So we generate a hadoop token file to pass kyuubi engine tokens to submit process.
+    // TODO: Removed this after FLINK-35525, delegation tokens will be passed by `kyuubi` provider
```

Comment:
I'm hesitant to mix two approaches up, maybe we need a switch

Reply:
FLINK-35525 has not yet been introduced in stable version. the hadoop token file way is currently only effective one.

Reply:
Flink 1.20 is on the way.

We can simply make the decision based on the Flink version, but considering vendors are likely to backport patches to their internal distributions, an explicit switch is preferred.

Reply:
> an explicit switch is preferred.

Do you mean add a configuration to control this behavior?

Reply:
Yes, a new configuration. Given the current approach is hacky and will be eventually removed, we can mark the configuration as internal.